### ISSUE TITLE
Node dependencies caming Corrupted

### ISSUE THREAD

**Issue by DanielProgrammer64** [expo/expo-cli#4738]
### Summary

That Issue is based on some dependencies just ggetting bugged with no reason, we create the expo web, and a lot of node_modules settings came corrupted ... 

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 20.2.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 9.6.6 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.70.8 => 0.70.8
      react-native-web: ~0.18.10 => 0.18.12
    Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

node: v20.2.0.

### Error output

WARNING: expo-cli has not yet been tested against Node.js v20.2.0.
If you encounter any issues, please report them to https://github.com/expo/expo-cli/issues

expo-cli supports following Node.js versions:
* >=12.13.0 <13.0.0 (Maintenance LTS)
* >=14.0.0 <15.0.0 (Active LTS)
* >=15.0.0 <17.0.0 (Current Release)

┌─────────────────────────────────────────────────────────────────────────┐
│                                                                         │
│   There is a new version of expo-cli available (6.3.10).                │
│   You are currently using expo-cli 4.13.0                               │
│   Install expo-cli globally using the package manager of your choice;   │
│   for example: `npm install -g expo-cli` to get the latest version      │
│                                                                         │
└─────────────────────────────────────────────────────────────────────────┘
┌───────────────────────────────────────────────────────────────────────────┐
│                                                                           │
│   This version of expo-cli is not supported anymore.                      │
│   It's highly recommended to update to the newest version.                │
│                                                                           │
│   The API endpoints used in this version of expo-cli might not exist,     │
│   any interaction with Expo servers may result in unexpected behaviour.   │
│                                                                           │
└───────────────────────────────────────────────────────────────────────────┘
Starting project at C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone
Developer tools running on http://localhost:19002
Opening developer tools in the browser...
Some dependencies are incompatible with the installed expo package version:
 - expo-status-bar - expected version: ~1.2.0 - actual version installed: 1.4.4
Your project may not work correctly until you install the correct versions of the packages.
To install the correct versions of these packages, please run: expo install [package-name ...]
Starting Metro Bundler
Missing package "metro/src/lib/attachWebsocketServer" in the project at: C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone
This usually means `react-native` is not installed. Please verify that dependencies in package.json include "react-native" and run `yarn` or `npm install`.
Error: Missing package "metro/src/lib/attachWebsocketServer" in the project at: C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone
This usually means `react-native` is not installed. Please verify that dependencies in package.json include "react-native" and run `yarn` or `npm install`.
    at resolveFromProject (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\importMetroFromProject.ts:21:11)
    at importFromProject (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\importMetroFromProject.ts:27:18)
    at Object.importMetroLibAttachWebsocketServerFromProject (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\importMetroFromProject.ts:49:10)
    at Object.createDevServerAsync (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\createDevServerAsync.ts:63:33)
    at runMetroDevServerAsync (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\MetroDevServer.ts:102:22)
    at startDevServerAsync (C:\Users\ProgramaçãoComDaniel\node_modules\xdl\src\start\startDevServerAsync.ts:76:49)
    at startAsync (C:\Users\ProgramaçãoComDaniel\node_modules\xdl\src\start\startAsync.ts:74:41)

### Reproducible demo or steps to reproduce from a blank project

› Choose an app to open your project at http://192.168.15.26:19000/_expo/loading
› Metro waiting on exp://192.168.15.26:19000
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Press a │ open Android
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
Starting Webpack on port 19006 in development mode.
‼ ｢wds｣: transportMode is an experimental option, meaning its usage could potentially change without warning
node:internal/crypto/hash:69
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:69:19)
    at Object.createHash (node:crypto:138:10)
    at module.exports (C:\Users\ProgramaçãoComDaniel\node_modules\webpack\lib\util\createHash.js:135:53)
    at NormalModule._initBuildHash (C:\Users\ProgramaçãoComDaniel\node_modules\webpack\lib\NormalModule.js:417:16)
    at handleParseError (C:\Users\ProgramaçãoComDaniel\node_modules\webpack\lib\NormalModule.js:471:10)
    at C:\Users\ProgramaçãoComDaniel\node_modules\webpack\lib\NormalModule.js:503:5
    at C:\Users\ProgramaçãoComDaniel\node_modules\webpack\lib\NormalModule.js:358:12
    at C:\Users\ProgramaçãoComDaniel\node_modules\loader-runner\lib\LoaderRunner.js:373:3
    at iterateNormalLoaders (C:\Users\ProgramaçãoComDaniel\node_modules\loader-runner\lib\LoaderRunner.js:214:10)
    at Array.<anonymous> (C:\Users\ProgramaçãoComDaniel\node_modules\loader-runner\lib\LoaderRunner.js:205:4)
    at Storage.finished (C:\Users\ProgramaçãoComDaniel\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:55:16)
    at C:\Users\ProgramaçãoComDaniel\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:91:9
    at C:\Users\ProgramaçãoComDaniel\node_modules\graceful-fs\graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v20.2.0
PS C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone> npx expo install react-native@0.70.8
› Installing 1 SDK 47.0.0 compatible native module using npm
> npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: instaclone@1.0.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"18.1.0" from react-native@0.70.8
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.70.8" from the root project
npm ERR!
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\ProgramaçãoComDaniel\AppData\Local\npm-cache\_logs\2023-07-24T02_48_27_605Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: C:\Users\ProgramaçãoComDaniel\AppData\Local\npm-cache\_logs\2023-07-24T02_48_27_605Z-debug-0.log

Error: npm exited with non-zero code: 1
Error: npm exited with non-zero code: 1
    at ChildProcess.completionListener (C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone\node_modules\@expo\spawn-async\build\spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:626:26)
    at ChildProcess.emit (node:events:511:28)
    at cp.emit (C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone\node_modules\cross-spawn\lib\enoent.js:34:29)
    at maybeClose (node:internal/child_process:1098:16)
    at ChildProcess._handle.onexit (node:internal/child_process:304:5)
    ...
    at spawnAsync (C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone\node_modules\@expo\spawn-async\build\spawnAsync.js:17:21)
    at NpmPackageManager._runAsync (C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone\node_modules\@expo\package-manager\build\NodePackageManagers.js:133:51)
    at NpmPackageManager.installAsync (C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone\node_modules\@expo\package-manager\build\NodePackageManagers.js:59:20)        
    at NpmPackageManager.addWithParametersAsync (C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone\node_modules\@expo\package-manager\build\NodePackageManagers.js:72:2    at async installPackagesAsync (C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone\node_modules\@expo\cli\build\src\install\installAsync.js:82:5)
PS C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone> npm uninstall expo
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: instaclone@1.0.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"18.1.0" from react-native@0.70.8
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.70.8" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\ProgramaçãoComDaniel\AppData\Local\npm-cache\_logs\2023-07-24T02_48_44_638Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: C:\Users\ProgramaçãoComDaniel\AppData\Local\npm-cache\_logs\2023-07-24T02_48_44_638Z-debug-0.log
PS C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone> npm uninstall expo --force
npm WARN using --force Recommended protections disabled.
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-native@0.70.8
npm WARN Found: react@18.2.0
npm WARN   peer react@"^18.2.0" from react-dom@18.2.0
npm WARN   node_modules/react-dom
npm WARN     peer react-dom@"^17.0.2 || ^18.0.0" from react-native-web@0.18.12
npm WARN     node_modules/react-native-web
npm WARN     1 more (the root project)
npm WARN   4 more (react-native-web, react-shallow-renderer, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"18.1.0" from react-native@0.70.8
npm WARN node_modules/react-native
npm WARN   react-native@"0.70.8" from the root project
npm WARN
npm WARN Conflicting peer dependency: react@18.1.0
npm WARN node_modules/react
npm WARN   peer react@"18.1.0" from react-native@0.70.8
npm WARN   node_modules/react-native
npm WARN     react-native@"0.70.8" from the root project
added 61 packages, removed 813 packages, changed 194 packages, and audited 819 packages in 2m

34 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
PS C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone> npm install -g expo-cli
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated expo-cli@6.3.10: The global Expo CLI has been superseded by 'npx expo' and eas-cli –– Learn more: https://blog.expo.dev/the-new-expo-cli-f4250d8e3421

changed 1185 packages in 2m

106 packages are looking for funding
  run `npm fund` for details
PS C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone> npx expo start
WARNING: expo-cli has not yet been tested against Node.js v20.2.0.
If you encounter any issues, please report them to https://github.com/expo/expo-cli/issues

expo-cli supports following Node.js versions:
* >=12.13.0 <13.0.0 (Maintenance LTS)
* >=14.0.0 <15.0.0 (Active LTS)
* >=15.0.0 <17.0.0 (Current Release)

┌─────────────────────────────────────────────────────────────────────────┐
│                                                                         │
│   There is a new version of expo-cli available (6.3.10).                │
│   You are currently using expo-cli 4.13.0                               │
│   Install expo-cli globally using the package manager of your choice;   │
│   for example: `npm install -g expo-cli` to get the latest version      │
│                                                                         │
└─────────────────────────────────────────────────────────────────────────┘
┌───────────────────────────────────────────────────────────────────────────┐
│                                                                           │
│   This version of expo-cli is not supported anymore.                      │
│                                                                           │
│   The API endpoints used in this version of expo-cli might not exist,     │
│   any interaction with Expo servers may result in unexpected behaviour.   │
│                                                                           │
└───────────────────────────────────────────────────────────────────────────┘
Starting project at C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone
Developer tools running on http://localhost:19002
Opening developer tools in the browser...
Some dependencies are incompatible with the installed expo package version:
 - expo-status-bar - expected version: ~1.2.0 - actual version installed: 1.4.4
Your project may not work correctly until you install the correct versions of the packages.
To install the correct versions of these packages, please run: expo install [package-name ...]
Starting Metro Bundler
Missing package "metro/src/lib/attachWebsocketServer" in the project at: C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone
This usually means `react-native` is not installed. Please verify that dependencies in package.json include "react-native" and run `yarn` or `npm install`.
Error: Missing package "metro/src/lib/attachWebsocketServer" in the project at: C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone
This usually means `react-native` is not installed. Please verify that dependencies in package.json include "react-native" and run `yarn` or `npm install`.
    at resolveFromProject (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\importMetroFromProject.ts:21:11)
    at importFromProject (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\importMetroFromProject.ts:27:18)
    at Object.importMetroLibAttachWebsocketServerFromProject (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\importMetroFromProject.ts:49:10)
    at Object.createDevServerAsync (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\metro\createDevServerAsync.ts:63:33)
    at runMetroDevServerAsync (C:\Users\ProgramaçãoComDaniel\node_modules\@expo\dev-server\src\MetroDevServer.ts:102:22)
    at startDevServerAsync (C:\Users\ProgramaçãoComDaniel\node_modules\xdl\src\start\startDevServerAsync.ts:76:49)
    at startAsync (C:\Users\ProgramaçãoComDaniel\node_modules\xdl\src\start\startAsync.ts:74:41)
PS C:\Users\ProgramaçãoComDaniel\Desktop\DevJourney\projeto4 - InstaCloneWithReact\InstaClone> npx expo-env-info 
Need to install the following packages:
  expo-env-info@1.0.5
Ok to proceed? (y) y

  expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 20.2.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 9.6.6 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.70.8 => 0.70.8
      react-native-web: ~0.18.10 => 0.18.12
    Expo Workflow: managed

**Comment by FelipeRobles92** [comment]
Hello, did you manage to find the problem?
